src/warc2zim/content_rewriting/ds.py

```diff
@@ -256,14 +255,15 @@ def filter_by_bitrate(array, max_bitrate, mime):
 ]
 
 
-def build_domain_specific_rewriter(path: str, url_rewriter):
+def get_ds_rules(path: str):
     """
-    This build a JsRewriter with extra rules depending of the path.
+    This build a domain specifc Rewriter (given in rewriter_class), by passing a set of
+    extra rules to the rewriter if needed.
     """
 
     for rule in RULES:
-        for contain in rule.contains:
-            if contain in path:
-                return JsRewriter(url_rewriter, rule.rx_rules)
+        for pattern in rule.contains:
+            if pattern in path:
+                return rule.rx_rules
 
-    return JsRewriter(url_rewriter)
+    return None
```

Comment:
Why not returning an empty list? Would make return type unconditional

Reply:
Empty list returned.
But the conditional on the call site is mainly for performance purpose, no need to run a RxRewriter if there is no rules.

Reply:
Boolean test of empty list is false as well and as good in this case